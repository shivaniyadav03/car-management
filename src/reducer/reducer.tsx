
const InitialValue={
    numberOfLots:[{
        lotid:1,
        assignedTo: null
    },
    {
        lotid:2,
        assignedTo: null
    },
    {
        lotid:3,
        assignedTo: null
    },
    {
        lotid:4,
        assignedTo: null
    },
    {
        lotid:5,
        assignedTo: null
    }
]
};

const randomIntFromInterval = (min: number, max: number) => { 
    return Math.floor(Math.random() * (max - min + 1) + min)
}

 const reducer=(state:any=InitialValue,action:any)=>{
 switch(action.type){
    case 'ASSIGN_LOT':{
        const currentNumberOfLots: any[] = state?.numberOfLots
        const unAssignedLots:any[] = currentNumberOfLots.filter((item:any)=>{
            return item.assignedTo==null;
        });
        Object.keys(action.payload).map(item => {
            const storeInRandomIndex = () => {
                const randomIndex: number = randomIntFromInterval(0,unAssignedLots.length-1);
                if(unAssignedLots[randomIndex]?.assignedTo){
                    storeInRandomIndex()
                }else{
                    unAssignedLots[randomIndex].assignedTo = action.payload[item];
                    alert('Assign Lots:'+randomIndex);
                }
            }
            storeInRandomIndex();
        })
        unAssignedLots.map((item: any) => {
            const currentIndex: number = currentNumberOfLots.findIndex((x: any) => item.lotid == x.lotid);
            if(currentIndex !== -1){
                currentNumberOfLots[currentIndex].assignedTo = item.assignedTo
            }
        })
        return {
            ...state,
            numberOfLots: currentNumberOfLots
        }
    }
    case 'EXIT_LOT':{
        const currentNumberOfLots:any[]=state?.numberOfLots;
         const updatedLots: any[] = currentNumberOfLots.map((item:any,index:number)=>{
          if(item.lotid==action.payload){
            item.assignedTo=null;            
          }
          return item;
         });
         return{
             ...state,
             numberOfLots: updatedLots
         }
    }
    default:return {
        ...state
    }
 }
}
export default reducer;