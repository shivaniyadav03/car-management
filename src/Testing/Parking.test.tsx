import {render,cleanup} from '@testing-library/react';
import ParkingArea from '../Pages/parkingArea';
import {createStore} from 'redux';
import reducer from '../reducer/reducer';
import {Provider} from 'react-redux';


beforeEach(()=>{
    document.body.innerHTML="";
});
afterEach(()=>{
    cleanup();
});
const mockedData=jest.fn();
jest.mock('react-router-dom',()=>{
    const updateData=jest.requireActual('react-router-dom');
    return{
    __esModule:true,
    ...updateData,
    useLocation:()=>mockedData,
    useNavigate:()=>mockedData,
    useDispatch:()=>mockedData
    };
});
describe('To create Snapshots',()=>{
test('',()=>{
    const store=createStore(reducer);
    render(
        <Provider store={store}>
            <ParkingArea/>
        </Provider>
    );
expect(screen).toMatchSnapshot();
});
});