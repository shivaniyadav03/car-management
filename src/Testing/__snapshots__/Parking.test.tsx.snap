// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To create Snapshots  1`] = `Screen {}`;
